### PR TITLE
add notes for n27827

### PR DESCRIPTION
**Pull request by josibake**
These are the main areas I want to focus on, but I plan on revisiting them on Monday/Tuesday to add it a bit more detail, links, etc. I expect the PR to change a lot in the next 2 weeks, so I'm hesitant to get too specific on the code (e.g. links to specific commits), and I also am looking for more conceptual feedback on a few areas of the PR. Lemme what you think of the approach though. Happy to update if you guys think a different approach is better!

### CODE REVIEW

File: _posts/2023-08-02-#27827.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+commit:
+---
+
+## Notes
+
+This PR implements [BIP352: Silent Payments](https://github.com/bitcoin/bips/pull/1458) in the Bitcoin Core wallet. BIP352 is "a protocol for static payment addresses in Bitcoin without on-chain linkability of payments or a need for on-chain notifications."
+
+Before reviewing the PR, it is _strongly_ recommended that you read the [BIP](https://github.com/bitcoin/bips/pull/1458), as well as the [reference implementation](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/reference.py) and [test vectors](https://github.com/bitcoin/bips/blob/cfe0771a0408a2d2de278d4e95bb9a33bd1615b2/bip-0352/send_and_receive_test_vectors.json). This PR implements the base functionality: sending and receiving. Additional features and broader RPC support are left for follow up PRs. The PR is heavily based on a proof of concept PR, so this PR review club will be primarily focused on:
+
+* Does this PR correctly implement the BIP?
```
Comment (glozow) [inline _posts/2023-08-02-#27827.md:19]: You can break this down into more specific functionality and add links to the code that does it. In the questions section you can ask "how is this implemented and is it correct" eg `EncodeSilentDestination`, `VerifySilentPayment`, detecting whether a transaction is a silent payment to us, calculating the keys from the address, etc.

File: _posts/2023-08-02-#27827.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+2. Did you take a look at the reference implementation on the BIP and the unit tests? If so, does the python reference implement the BIP correctly? What test cases could be added?
+
+3. BIP352 allows sending and receiving to be implemented separately. As such, this PR is organized into three main sections: 1) adding the necessary cryptographic primitives, 2) implementing sending, and 3) implementing receiving. What are the advantages of this approach? Does it make sense to split this into multiple PRs? Why or why not?
+
+4. How does the PR create the spend and scan keys? Are there any issues with this approach?
+
+5. Is a descriptor the right approach here? Why or why not? If it is, how would you modify the `sp()` descriptor to correctly implement the BIP?
```
Comment (glozow) [inline _posts/2023-08-02-#27827.md:34]: Perhaps name some alternative approaches

**Review by glozow — COMMENTED** [review]
I don't think it should just be conceptual questions, as it's PR review club and the goal is to get people to look at the code and learn how it works. There are 2 meetings so hopefully it's not just surface-level and the more notes you provide, the deeper people are able to review the PR.

Even if the implementation isn't finalized, there's plenty of Bitcoin Core wallet code to learn here. Some ideas when you revisit on Monday/Tuesday:
- How descriptors are implemented and used in the wallet code
- What `ParseScript` is doing
- How `CTxDestination` differs from a scriptPubKey
- At a high-ish level, how `CKey`, `CPubkey`, `XOnlyPubKey`, `HashWriter`, etc. are used

**Comment by josibake** [comment]
hey @glozow , thanks for the review and suggestions! will incorporate your feedback and try to have this ready to go tomorrow by lunch (1500 UTC)

**Comment by josibake** [comment]
Ended up re-writing the first half of the PR and opening it as its own. Closing in favor of https://github.com/bitcoin-core-review-club/website/pull/705

**Pull request closed without merging**